---
title: "THRA Isoform 1 and Isoform 2 Relative Expression"
output:
  html_document:
    toc: True
    toc_float: True
    df_print: paged
---

INTRODUCTION to the Experiment

A total of 24 samples of human tissues. RNA seq data. First-Stranded pair-end reads. Read counts of exon 9a (Chr17:40089333) and exon 9b (Chr17:40089334)

```{r}
#load libraries
library(tidyverse)
library(readr)
library(ggplot2)
library(plotly)
library(matrixStats)
library(ggrepel)
library(scales)
library(readxl)
library(dplyr)

```


```{r}
## set paths for output figure
path_plots <- "~/OneDrive - Charité - Universitätsmedizin Berlin/Schuelke_Lab/EG15_RNA_Seq/TAKARA_Samples/THRA_isoform_Expression_Coverage"

## load metadata file

metadata <- read_excel("metadata.xlsx")


##Load bedtools outputs

file_links_bedtools <- list.files(path= "~/OneDrive - Charité - Universitätsmedizin Berlin/Schuelke_Lab/EG15_RNA_Seq/TAKARA_Samples/bedtools_outputs" , pattern = "*.txt", full.names=T)


# initialize an empty dataframe
data_bedtools <- data_frame("V1"=character(),
                   "V2"=integer(),
                   "V3"=integer(),
                   "V4"=character(),
                   "V5"=double(),
                   "Sample"=character()) 

for (x in file_links_bedtools) {
  table_sample <- read.delim(x, header = FALSE) ## read table
  basename_sample <- str_remove(basename(x), "_Aligned.out.bedtools.txt")  ## get the sample name from the file path
  table_sample <- mutate(table_sample, "Sample"=basename_sample)
  data_bedtools <- bind_rows(data_bedtools, table_sample)
  
}

data_bedtools

##Load mosdepth outputs

file_links_mosdepth <- list.files(path= "~/OneDrive - Charité - Universitätsmedizin Berlin/Schuelke_Lab/EG15_RNA_Seq/TAKARA_Samples/mosdepth_outputs" , pattern = "*.bed", full.names=T)


# initialize an empty dataframe
data_mosdepth <- data_frame("V1"=character(),
                   "V2"=integer(),
                   "V3"=integer(),
                   "V4"=character(),
                   "V5"=double(),
                   "Sample"=character()) 

for (x in file_links_mosdepth) {
  table_sample <- read.delim(x, header = FALSE) ## read table
  basename_sample <- str_remove(basename(x), "_Aligned.out.regions.bed")  ## get the sample name from the file path
  table_sample <- mutate(table_sample, "Sample"=basename_sample)
  data_mosdepth <- bind_rows(data_mosdepth, table_sample)
  
}

data_mosdepth

```

```{r}
## Let's rearrange the data in a useful way
data_mosdepth <- data_mosdepth %>%
  select(Sample, Isoform = V4, Reads_count =V5) %>%
  spread(key="Isoform", value="Reads_count") %>%
  mutate("Package" = as.factor("mosdepth")) 

data_mosdepth

data_bedtools <- data_bedtools %>%
  select(Sample, Isoform = V4, Reads_count =V5) %>%
  spread(key="Isoform", value="Reads_count")%>%
  mutate("Package" = as.factor("bedtools"))

data_bedtools
```

```{r}
## Let's merge the two dataframes into one

dataset <- full_join(data_mosdepth, data_bedtools)

dataset
```



```{r}
## Let's calkculate THRA1 (counts of 9b) and THRA2 (9a-9b)

dataset <- dataset %>%
  mutate("THRA1"= dataset$`9b`) %>%
  mutate("THRA2"=dataset$`9a`-dataset$`9b`) %>%
  rename("Read_counts_9a" = "9a") %>%
  rename("Read_counts_9b" = "9b") %>%
  relocate(Package, .after=Sample)

dataset

```
```{r}
## Let's add the final calculations

dataset_final <- dataset %>%
  mutate("Delta_A1-A2" = THRA1 - THRA2) %>%
  mutate("THRA1_higher" = THRA1 > THRA2)

dataset_final
  
```






```{r}
### Lets compare the read counts of exon 9a between the 2 tools
graph_data <- dataset_final %>%
  select(Sample, Package, Read_counts_9a)


ggplot(graph_data, aes(x = Sample, y = Read_counts_9a, fill = Package)) +
  geom_col( width=0.5, position=position_dodge(width=0.7)) +
  scale_y_continuous("Read counts") +
  scale_fill_manual("", values = c("bedtools" = "darksalmon", "mosdepth" = "#56B4E9")) +
  ggtitle("Tools comparison - Exon 9a") +
  theme_light(base_size = 12) +
  scale_x_discrete(guide = guide_axis(angle = -35)) +
  theme(axis.title.x=element_blank())

#ggplotly()
```

```{r}
### Lets compare the read counts of exon 9b between the 2 tools
graph_data <- dataset_final %>%
  select(Sample, Package, Read_counts_9b)


ggplot(graph_data, aes(x = Sample, y = Read_counts_9b, fill = Package)) +
  geom_col( width=0.5, position=position_dodge(width=0.7)) +
  scale_y_continuous("Read counts") +
  scale_fill_manual("", values = c("bedtools" = "darksalmon", "mosdepth" = "#56B4E9")) +
  ggtitle("Tools comparison - Exon 9b") +
  theme_light(base_size = 12) +
  scale_x_discrete(guide = guide_axis(angle = -35)) +
  theme(axis.title.x=element_blank())

#ggplotly()
```



```{r}
### Lets graph the results
graph_data <- dataset_final %>%
  filter(Package == "mosdepth") %>%
  select(Sample, "THRA1"=THRA1, "THRA2"=THRA2)%>%
  pivot_longer(!Sample, names_to = "Detector", values_to = "Reads_count")



ggplot(graph_data, aes(x = Sample, y = Reads_count, fill = Detector)) +
  geom_col( width=0.5, position=position_dodge(width=0.7)) +
  scale_y_continuous("Count of reads") +
  scale_fill_manual("", values = c("THRA1" = "darksalmon", "THRA2" = "#56B4E9")) +
  ggtitle("THRA Isoform 1 and Isoform 2 Relative Expression") +
  theme_light(base_size = 12) +
  scale_x_discrete(guide = guide_axis(angle = -35)) +
  theme(axis.title.x=element_blank())

#ggplotly()
```


```{r}
### Lets plot the difference of the percentage

differential <- metadata %>%
  mutate(delta_qPCR=THRA1-THRA2)%>%
  select(Sample, delta_qPCR)%>%
  arrange(desc(delta_qPCR)) %>%
  mutate( order= row_number())

  ggplot(differential, aes(x = order, y = delta_qPCR)) +
  geom_point() +
  geom_hline(yintercept=0, linetype="dashed", color = "black")+
  scale_x_discrete("") +
  ggtitle("THRA isoform expression pattern (THRA1/THRA2)") +
  theme_light() +
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank()) +
  geom_segment(aes(xend=order,yend=0),linetype="dashed", color = "red")+
   geom_text(aes(label=Sample),hjust=0, vjust=-1.0, angle = +35)+
  annotate("text", x = 12, y = 500, label = "Higher THRA1") +
  annotate("text", x = 12, y = -400, label = "Higher THRA2")+
  scale_y_continuous(limits=c(-500, 500), expand=c(0.1,4)) 




 ggplotly()   

```

```{r}
### heatmap with continuos delta THRA1vsA2
ggplot(differential, aes(1, reorder(Sample,delta_qPCR), fill=delta_qPCR)) +
  geom_tile()+
  ggtitle("THRA isoform expression pattern (THRA1/THRA2)") +
  theme_light(base_size = 12)+
  theme(axis.title.y=element_blank(),
        axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank()) +
  scale_fill_gradient2('delta_reads', limits=c(-350, 350), breaks = c( -350, -150, 0, 150, 350),  low = "#56B4E9", high = "darksalmon", guide="colorbar")

ggsave("Heatmap_THRA1vsTHRA2_RNA_Seq.png" , device=png, dpi = 600, bg = "transparent", width = 20, height = 15, units = "cm" )
```


```{r}
### heatmap with UP or DOWN visualization
comparison_heatmap_up_down <- differential %>%
  mutate(THRA1vsTHRA2=ifelse(delta_qPCR>0 , "Higher" , "Lower"))
  


ggplot(comparison_heatmap_up_down, aes(1, reorder(Sample,delta_qPCR), fill=THRA1vsTHRA2)) +
  geom_tile()+
  ggtitle("THRA isoform expression pattern (THRA1/THRA2)") +
  theme_light(base_size = 12)+
  theme(axis.title.y=element_blank(),
        axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank()) +
  scale_fill_manual(values = c("darksalmon", "#56B4E9"))

ggsave("Heatmap_THRA1vsTHRA2_RNA_Seq_aut_aut.png" , device=png, dpi = 600, bg = "transparent", width = 20, height = 15, units = "cm" )
```





```{r}
###Lets compare the methods


### heatmap with continuos delta
ggplot(dataset_final, aes(Package, reorder(Sample,`Delta_A1-A2`) , fill=`Delta_A1-A2`)) +
  geom_tile()+
  ggtitle("Delta comparison between methods (THRA1/THRA2)") +
  theme_light(base_size = 12)+
  theme(axis.title.y=element_blank()) +
  theme(axis.title.x=element_blank()) +
  scale_fill_gradient2('delta', breaks = c(-230, 0,  250),  low = "#56B4E9", high = "darksalmon", guide="colorbar")
```

```{r}
## save
ggsave("Heatmap_methods_comparison.png" , device=png, dpi = 600, path = path_plots, bg = "transparent", width = 20, height = 15, units = "cm" )
```

